src/lib.rs

```diff
@@ -412,6 +420,50 @@ fn run_for_context<TContext: Sync>(
     })
 }
 
+pub fn run_for_slice_with_callback(
+    slice: &[u8],
+    args: Args,
+    mut callback: impl FnMut(CaptureInfo) + Sync,
+) -> Result<RunStatus, Error> {
+    let language = args.language.ok_or(Error::LanguageMissingForSlice)?;
+    let query_text = args.get_loaded_query_text()?;
```

Comment:
Don't know if there are great ways to DRY this up more wrt other entry points, didn't worry about that too much at the moment